stdNullPtr/Dynamic-Email-Generator
# Secure API Communication - TLS

## Overview
Added TLS configuration to the API, allowing only HTTPS traffic.
Configured Nginx to serve requests to the new HTTPS endpoints.
Added Integration tests for the controller.
Updated Github workflow.
Added a script to easily create a sample keystore.
Updated documentation.

## Related Issues
Secure communication #18 - Solves the first part of the issue - forcing HTTPS traffic.

## Testing Instructions
1. Checkout the branch and build the project with ```mvn clean verify```.
2. Start the service as a standalone, locally, without docker.
    * Follow the new instructions described in the README, as part of this PR, for setting up a certificate
    * Start the spring boot app with the following environment variables (assuming you followed the instructions and used default keystore generation)
    KEYSTORE_PATH=./tls/keystore/springboot.p12;KEYSTORE_PASSWORD=123456
3. Hit the swagger page through HTTPS (--insecure because our certificate is self-signed):
    * ```curl -v --insecure https://localhost:8443/swagger-ui/index.html```  
    ![Screenshot 2024-05-09 122844](https://github.com/stdNullPtr/Dynamic-Email-Generator/assets/8255033/154a01d4-19f6-4e17-bfb8-e3861a9e327e)
